### PR TITLE
Feature/fix groupid type

### PR DESCRIPTION
**Pull request by giovannifilannino**
Updated GroupId data type.

### Checklist
- [ ] I've run `gradle junitPlatformTest` from the root directory to see all new and existing tests pass
- [X] I've followed the _harvest-client_ code style
- [X] I've read the [Contribution Guidelines][contribution guidelines]
- [X] I've updated the documentation if necessary.

### Motivation and Context
The ExternalReference's groupId is mapped as Long but during some tests I received a String.

### Description
Changed from Long to String.

[contribution guidelines]: ../CONTRIBUTING.md

### CODE REVIEW

File: src/main/java/ch/aaap/harvestclient/domain/ExternalService.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @Value.Immutable
 public interface ExternalService extends Reference<ExternalService> {
 
-    Long getGroupId();
+    String getGroupId();
```
Comment (mnembrini) [inline src/main/java/ch/aaap/harvestclient/domain/ExternalService.java]: Good change but it could use a test -> see discussion on slack

**Comment by robertblust** [comment]
@mnembrini i guess i got a notification about something you posted on slack, but can not find it anymore. In which channel you posted?

**Comment by mnembrini** [comment]
> @mnembrini i guess i got a notification about something you posted on slack, but can not find it anymore. In which channel you posted?

I guess this is confusing,  I wrote directly to @giovannifilannino to explain how to write a test

File: src/test/java/ch/aaap/harvestclient/impl/timesheet/TimesheetsApiListTest.java [hunk 2]
```diff
@@ -346,4 +346,29 @@ void listByDateRange() {
...
+            .userReference(user)
+            .externalReference(externalService)
+            .build();
+
+        timeEntry = api.create(creationInfo);
+        TimeEntryFilter filter = new TimeEntryFilter();
+        filter.setTo(LocalDate.of(2000,1,6));
+        List<TimeEntry> timeEntries = api.list(filter);
+        TimeEntry timeEntry = timeEntries.stream().filter(te -> te.getExternalService() != null).findAny().orElse(null);
+        assertThat(timeEntry != null && timeEntry.getExternalService() != null && timeEntry.getExternalService().getGroupId().equalsIgnoreCase("sample"));
```
Comment (mnembrini) [inline src/test/java/ch/aaap/harvestclient/impl/timesheet/TimesheetsApiListTest.java]: assertThat is used for fluent assertions, you have to assert something after it.
Here you want assertTrue or you need to use assertThat properly, at the moment this assert is not functional

**Pull request merged**